lib/exprengine.cpp

```diff
@@ -79,7 +79,7 @@ namespace {
 
         void print() {
             std::set<std::pair<int,int>> locations;
-            for (auto it : map) {
+            for (const auto& it : map) {
```

Comment:
doesn't mean that an extra load is needed to read `it.first` or `it.second`... first load the reference to see what it references..   I thought that `it` was a pretty trivial value to read from the map?

Reply:
II don't think reading `it` changes with this - but the loop would copy the pair on each run.

Reply:
the pair is 2 ints.. 8 bytes .. the reference is 8 bytes also. so I guess the runtime cost for the read would be the same if the compiler can optimise 2 4-byte reads into 1 8-byte read.

Reply:
I checked these with clang-tidy enabled and it reports this. So I consider keeping this change.